compiler/src/steps/add-examples.ts

```diff
@@ -0,0 +1,261 @@
+    const folders = entries
+      .filter(entry => entry.isDirectory())
+      .map(entry => entry.name)
+    return folders
+  }
+
+  // Get all the files in a folder.
+  getFilesInFolder (folderPath: string): string[] {
+    const entries = fs.readdirSync(folderPath, { withFileTypes: true })
+    const files = entries
+      .filter(entry => entry.isFile())
+      .map(entry => entry.name)
+    return files
+  }
+
+  // Check if a path exists and is a directory.
+  isDirectory (path: string): boolean {
+    try {
+      const stats = fs.statSync(path)
+      return stats.isDirectory()
+    } catch (error) {
+      if (error.code === 'ENOENT') {
+        // Path does not exist
+        return false
+      } else {
+        // Other error, rethrow
+        throw error
+      }
+    }
+  }
+
+  // Given the spec location of a request or response,
+  // return the path to the examples folder for that
+  // request or response.
+  getExamplesFolder (specLocation: string): string | undefined {
+    const specDir = path.dirname(specLocation)
+    const specPath = path.join(this.specsFolder, specDir)
+    const examplesFolder = path.join(specPath, 'examples')
+    if (this.isDirectory(examplesFolder)) {
+      return examplesFolder
+    }
+    return undefined
+  }
+
+  // Given an examples request or response folder, return all the
+  // valid example files in that folder.
+  getExampleFiles (folder: string): string[] {
+    if (!this.isDirectory(folder)) {
+      return []
+    }
+    // Currently we only allow YAML example files.
+    const exampleFiles = this.getFilesInFolder(folder)
+      .filter(file => file.endsWith('.yml') || file.endsWith('.yaml'))
+    if (exampleFiles.length === 0) {
+      this.warning(`No example files found in ${folder}`)
+      return []
+    }
+    return exampleFiles
+  }
+
+  // Look up all the example files in a folder. Use the filename without extension
+  // as the name of the example, and the YAML content as the example value.
+  // Return a map of example names to example values.
+  getExampleMap (folder: string): Map<string, model.Example> {
+    const exampleFiles = this.getExampleFiles(folder)
+    const examples = new Map<string, model.Example>()
+    for (const fileName of exampleFiles) {
+      const filePath = path.join(folder, fileName)
+      const exampleFileContent = fs.readFileSync(filePath, 'utf8')
+      const exampleName = path.basename(fileName, path.extname(fileName))
+      const example: model.Example = yaml.load(exampleFileContent)
+      // Some of the example files set their 'value' as a JSON string,
+      // and some files set their 'value' as an object. For consistency,
+      // if the value is not a JSON string, convert it to a JSON string.
+      if (typeof example.value !== 'string') {
+        // Convert to prettified JSON string
+        example.value = JSON.stringify(example.value, null, 2)
+      }
+      examples[exampleName] = example
```

Comment:
When using a `Map` this should be:

```suggestion
      examples.set(exampleName, example)
```

Reply:
Fixed in 7f3db23d55e5b38b551c51a698e7d67d6fc132ca